fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/api/RunreportsApiResource.java

```diff
@@ -70,6 +72,29 @@ public RunreportsApiResource(final PlatformSecurityContext context, final ReadRe
         this.reportingProcessServiceProvider = reportingProcessServiceProvider;
     }
 
+    @GET
+    @Path("/availableExports/{reportName}")
+    @Consumes({ MediaType.APPLICATION_JSON })
+    @Produces({ MediaType.APPLICATION_JSON })
+    @Operation(summary = "Return all available export types for the specific report", description = "Returns the list of all available export types.")
+    @ApiResponses({
+            @ApiResponse(responseCode = "200", description = "", content = @Content(array = @ArraySchema(schema = @Schema(implementation = ReportExportType.class)))) })
+    public Response retrieveAllAvailableExports(@PathParam("reportName") @Parameter(description = "reportName") final String reportName,
+            @Context final UriInfo uriInfo,
+            @DefaultValue("false") @QueryParam(IS_SELF_SERVICE_USER_REPORT_PARAMETER) @Parameter(description = IS_SELF_SERVICE_USER_REPORT_PARAMETER) final boolean isSelfServiceUserReport) {
+        MultivaluedMap<String, String> queryParams = new MultivaluedStringMap();
+        queryParams.putAll(uriInfo.getQueryParameters());
+
+        final boolean parameterType = ApiParameterHelper.parameterType(queryParams);
+        String reportType = this.readExtraDataAndReportingService.getReportType(reportName, isSelfServiceUserReport, parameterType);
+        ReportingProcessService reportingProcessService = this.reportingProcessServiceProvider.findReportingProcessService(reportType);
+        if (reportingProcessService == null) {
```

Comment:
Let's not put this logic here but in the provider.

Reply:
After discuss, currently not changed, because used by another method.